tests/coords/transform_coords_test.py

```diff
@@ -175,11 +207,149 @@ def abcd(ac, bd):
+    original = sc.DataArray(data=a + c, coords={'a': a, 'c': c, 'e': a})
+    da = original.transform_coords(['f'], graph=graph)
+    assert da.dims == ['b', 'c']
+
+    # e is a dimension coord
+    e = a.rename_dims({'a': 'e'})
+    original = sc.DataArray(data=a + e, coords={'a': a, 'c': a, 'e': e})
+    da = original.transform_coords(['f'], graph=graph)
+    assert da.dims == ['b', 'e']
+
+
+def test_dim_rename_produced_dim_coord():
+    # c is a dimension coordinate even though it is computed from a and b.
+    # *a    b
+    #   \  /
+    #    *c
+    #     |
+    #    *d
+    def f_c(a, b):
+        return c
+
+    graph = {'c': f_c, 'd': 'c'}
+    original = sc.DataArray(data=a + c, coords={'a': a, 'b': a})
+    da = original.transform_coords(['d'], graph=graph)
+    assert da.dims == ['a', 'd']
+
+    # b is a dim coord
+    graph = {'c': f_c, 'd': 'c'}
+    original = sc.DataArray(data=a + b + c, coords={'a': a, 'b': b})
+    da = original.transform_coords(['d'], graph=graph)
+    assert da.dims == ['a', 'b', 'd']
+
+
+def test_dim_rename_produced_dim_coord_cycle():
+    # c is a dimension coordinate even though it is computed from a.
+    #    a
+    #   / \
+    # *c   b
+    #   \ /
+    #   *d
+    def f_c(a):
+        return c
+
+    def f_d(b, c):
+        return b + c
+
+    graph = {'c': f_c, 'b': 'a', 'd': f_d}
+    original = sc.DataArray(data=c, coords={'a': a.rename_dims({'a': 'c'})})
+    da = original.transform_coords(['d'], graph=graph)
+    assert da.dims == ['d']
+
+    # a is a dim coord
+    original = sc.DataArray(data=a + c, coords={'a': a})
+    da = original.transform_coords(['d'], graph=graph)
+    assert da.dims == ['a', 'c']
+
+
+@pytest.mark.parametrize('keep_inputs', (True, False))
+def test_dim_rename_keep_arguments(keep_inputs):
+    # *x
+    #  |
+    #  y
+    #  |
+    #  a   *b
+    #   \  |
+    #    ab
+    def y(x):
+        return x
+
+    graph = {'y': y, 'a': 'y', 'ab': ab}
+    x = a.rename_dims({'a': 'x'})
+    original = sc.DataArray(data=x + b, coords={'x': x, 'b': b})
+
+    da = original.transform_coords(['ab'],
+                                   graph=graph,
+                                   keep_intermediate=True,
+                                   keep_aliases=True,
+                                   keep_inputs=keep_inputs)
+    assert da.dims == ['a', 'b']
+    da = original.transform_coords(['ab'],
+                                   graph=graph,
+                                   keep_intermediate=False,
+                                   keep_aliases=True,
+                                   keep_inputs=keep_inputs)
+    assert da.dims == ['a', 'b']
+    da = original.transform_coords(['ab'],
+                                   graph=graph,
+                                   keep_intermediate=True,
+                                   keep_aliases=False,
+                                   keep_inputs=keep_inputs)
+    assert da.dims == ['y', 'b']
```

Comment:
Not sure this makes sense. I don't feel the options to keep aliases, intermediates, or inputs should have an effect on the end result. This feels confusing and is bound to produce bugs? What is the motivation for this?

Reply:
The motivation is that graphs can be black boxes when they are predefined and maybe even wrapped up in a function like `scn.convert`. In that case, suppose you compute Q but for some reason, the tof dim cannot be renamed to Q and ends up being renamed to wavelength. But wavelength is nowhere in the output because `keep_intermediates=False`. So that dimension name makes no sense to the user.

Reply:
Ok, I still feel it is a bit odd, but maybe not a major issue. Would changing this later be relatively easy, or does it significantly affect the implementation?

Reply:
It is essentially a single if-statement that could be removed at any time.